runnable/generate_request_minerstatistics.py

```diff
@@ -0,0 +1,161 @@
+        ledger=filtered_ledger,
+        evaluation_time_ms=time_now,
+        decay_coefficient=risk_adjusted_decay_coefficient
+    )
+
+    for hotkey, miner_ledger in returns_ledger.items():
+        scoringunit = ScoringUnit.from_perf_ledger(miner_ledger)
+        augmented_return_cps[hotkey] = Scoring.return_cps(scoringunit)
+
+    for hotkey, miner_ledger in risk_adjusted_ledger.items():
+        scoringunit = ScoringUnit.from_perf_ledger(miner_ledger)
+        augmented_omega_cps[hotkey] = Scoring.omega_cps(scoringunit)
+        augmented_inverted_sortino_cps[hotkey] = Scoring.inverted_sortino_cps(scoringunit)
+
+    ## This is when we only want to look at the successful miners
+    successful_ledger = subtensor_weight_setter.filtered_ledger(hotkeys=challengeperiod_success_hotkeys)
+    checkpoint_results = Scoring.compute_results_checkpoint(successful_ledger, evaluation_time_ms=time_now)
+    challengeperiod_scores = [ (x, ValiConfig.SET_WEIGHT_MINER_CHALLENGE_PERIOD_WEIGHT) for x in challengeperiod_testing_hotkeys ]
+    scoring_results = checkpoint_results + challengeperiod_scores
+    
+    final_dict = {
+        'version': ValiConfig.VERSION,
+        'created_timestamp_ms': time_now,
+        'created_date': TimeUtil.millis_to_formatted_date_str(time_now),
+        'weights': dict(scoring_results),
+        'penalties': {
+            "consistency": consistency_penalties,
+        },
+        "metrics": {
+            "omega_cps": omega_cps,
+            "inverted_sortino_cps": inverted_sortino_cps,
+            "return_cps": return_cps
+        },
+        "augmented_metrics": {
+            "augmented_omega_cps": augmented_omega_cps,
+            "augmented_inverted_sortino_cps": augmented_inverted_sortino_cps,
+            "augmented_return_cps": augmented_return_cps
+        },
+        "constants":{
+            "set_weight_lookback_range_days": ValiConfig.SET_WEIGHT_LOOKBACK_RANGE_DAYS,
+            "set_weight_minimum_positions": ValiConfig.SET_WEIGHT_MINIMUM_POSITIONS,
+            "set_weight_minimum_position_duration_ms": ValiConfig.SET_WEIGHT_MINIMUM_POSITION_DURATION_MS,
+            "omega_minimum_denominator": ValiConfig.OMEGA_MINIMUM_DENOMINATOR,
+            "omega_ratio_threshold": ValiConfig.OMEGA_RATIO_THRESHOLD,
+            "probabilistic_sharpe_ratio_threshold": ValiConfig.PROBABILISTIC_LOG_SHARPE_RATIO_THRESHOLD,
+            "annual_risk_free_rate": ValiConfig.ANNUAL_RISK_FREE_RATE,
+            "lookback_range_days_risk_free_rate": ValiConfig.LOOKBACK_RANGE_DAYS_RISK_FREE_RATE,
+            "max_total_drawdown": ValiConfig.MAX_TOTAL_DRAWDOWN,
+            "max_daily_drawdown": ValiConfig.MAX_DAILY_DRAWDOWN,
+        },
+    }
+
+    if write_validator_checkpoint:
+        output_file_path = ValiBkpUtils.get_vali_outputs_dir() + "minerstatistics.json"
+        ValiBkpUtils.write_file(
+            output_file_path,
+            final_dict,
+        )
+
+    if write_legacy:
```

Comment:
you can remove "write legacy" logic. Don't need to gate with a flag "write_validator_checkpoint" either.